### PR TITLE
[ANOMALY SEGMENTATION] Use only polygon annotations for evaluation

### PR DESCRIPTION
**Pull request by djdameln**
This PR fixes an issue in anomaly segmentation where the evaluation was done on both global (image-level) and local (polygon) annotations. With these changes, the resultset is now split into a subset that has local annotations available and a subset that has only global annotations available. If the local subset contains anomalous images, it will be used to compute performance. Otherwise the global dataset will be used (in this case the reported performance will be image-level F1). This would allow users to train the task and see the predictions before making any polygon annotations.

Additionally, a similar split is done for the validation set before sending the data to Anomalib. This makes sure that Anomalib does not try to compute segmentation performance without ground truth masks available. If the global dataset is used for validation, Anomalib will use the image-level threshold to generate segmentation predictions.

[BMM](https://ci.iotg.sclab.intel.com/job/IMPT/job/IMPTOps/27001/)

### CODE REVIEW

**Comment by djdameln** [comment]
run ote_sdk tests

**Comment by Ilya-Krylov** [comment]
run ote_sdk tests

**Comment by Ilya-Krylov** [comment]
run ote_sdk tests

**Comment by Ilya-Krylov** [comment]
run ote_sdk tests

**Comment by Ilya-Krylov** [comment]
http://213.221.44.203/job/ote/job/precommit/job/ote_sdk_tests_runner/753/console 
so can think that tests passed, there was no problem with anomaly

**Comment by djdameln** [comment]
Verified that the changes achieve the desired effect in the UI. I think this can be merged @Ilya-Krylov
